gputop-csv/gputop-csv.js

```diff
@@ -456,19 +515,42 @@ function write_rows(metric, accumulator)
     }
 }
 
+var flag = 0;
+
 GputopCSV.prototype.notify_accumulator_events = function(metric, accumulator, events_mask) {
     if (events_mask & 1) //period elapsed
         this.accumulator_clear(accumulator);
 
     if (this.write_queued_)
         return;
 
-    setTimeout(() => {
-        this.write_queued_ = false;
-        write_rows.call(this, metric, accumulator);
-    }, 0.2);
+    if (this.idle_flag < 4) {
+        flag = 0;
+        setTimeout(() => {
+            this.write_queued_ = false;
+            write_rows.call(this, metric, accumulator);
+        }, 0.2);
 
-    this.write_queued_ = true;
+        this.write_queued_ = true;
+    } else {
+        if (flag === 0) {
+            flag = 1;
+            if (this.pretty_print_csv_)
+                stderr_log.error("No context is running on this vGPU now");
+            else
+                this.stream.write("No context is running on this vGPU now\n");
+        }
+
+        for (var c = 0; c < this.counters_.length; c++) {
+            var counter = this.counters_[c];
+            if (counter.record_data === true) {
+                var accumulated_counter =
+                    accumulator.accumulated_counters[counter.cc_counter_id_];
+                n_rows = 2;
```

Comment:
What's going on here? Why bump the number of rows to 2?

Reply:
Sorry for my mistake, I have removed this line.